File: backend/Model/responseModel.js

const mongoose = require('mongoose')


const Schema = mongoose.Schema
const responseSchema = new Schema({
    
    driver_id: {
        type: Schema.Types.ObjectId,
        required: true,
    },

    manager_id: {
        type: Schema.Types.ObjectId, 
        required: true
    },

    content: {
        type: String,
        required: true
    },
    issue_id: {
      type: Schema.Types.ObjectId,
      required: true,
    }
    
}, {timestamps: true})

module.exports = mongoose.model('Response', responseSchema)

File: flutter_app/lib/issue/models/responseModel.dart
class Response {
  final String id;
  final String driverId;
  final String managerId;
  final String content;
  final String issueId;

  Response(
      {required this.id,
      required this.driverId,
      required this.managerId,
      required this.content,
      required this.issueId});

  // factory Response.fromJson(Map<String, dynamic> json) {
  //   return Response(
  //     id: json['id'],
  //     name: json['name'],
  //     description: json['description'],
  //   );
  // }

  Map<String, dynamic> toJson() {
    return {
      'driver_id': driverId,
      'manager_id': managerId,
      'content': content,
      'issue_id': issueId,
    };
  }
}
